xcode/Subconscious/Shared/Services/GatewayProvisioningService.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  GatewayProvisioningService.swift
+//  Subconscious
+//
+//  Created by Ben Follington on 1/5/2023.
+//
+
+import Foundation
+import Combine
+import os
+
+struct ProvisionGatewayRequest: Codable {
+    var invite_code: String
+    var sphere: String
+}
+
+struct ProvisionGatewayErrorResponse: Codable {
+    var error: String?
+}
+
+struct ProvisionGatewayResponse: Codable {
+    var gateway_id: String
+}
+
+struct ProvisionGatewayStatusResponse: Codable {
+    var status: String
+    var did: String?
+    var url: String?
+}
+
+enum GatewayProvisioningServiceError: Error {
+    case failedToProvisionGateway(String)
+    case failedToCheckGatewayProvisioningStatus(String)
+    case gatewayIsNotReady
+    case invalidStatusResponse(ProvisionGatewayStatusResponse)
+}
+
+extension GatewayProvisioningServiceError: LocalizedError {
+    var errorDescription: String? {
+        switch self {
+        case .failedToProvisionGateway(let contentType):
+            return String(
+                localized: "Failed to provision gateway: \(contentType)",
+                comment: "GatewayProvisioningService error description"
+            )
+        case .failedToCheckGatewayProvisioningStatus(let contentType):
+            return String(
+                localized: "Failed to check gateway status: \(contentType)",
+                comment: "GatewayProvisioningService error description"
+            )
+        case .gatewayIsNotReady:
+            return String(
+                localized: "Gateway is not provisioned yet",
+                comment: "GatewayProvisioningService error description"
+            )
+        case .invalidStatusResponse(let res):
+            return String(
+                localized: """
+                    Invalid status response. \
+                    (status=\(res.status), \
+                    did=\(res.did ?? "<none>"), \
+                    url=\(res.url ?? "<none>"))
+                    """,
```

Comment:
TIL the closing `"""` determined the relative level of indentation captured within the string: https://www.swiftbysundell.com/tips/multiline-string-literal-tips-and-tricks/